plugins/BaseKillPlugin.cpp

```diff
@@ -58,10 +59,17 @@ bool BaseKillPlugin::tryToKillCgroup(
     if (nr_killed == last_nr_killed) {
       break;
     }
-    last_nr_killed = nr_killed;
-
-    // give it a breather before killing again
-    std::this_thread::sleep_for(1s);
+    OOMD_SCOPE_EXIT {
+      last_nr_killed = nr_killed;
```

Comment:
Why not just move this to the end of the block?  There aren't any other ways to leave the block between here and the `}`.